Fix UI callback `crit_load` check for Custom Generation Battery configs

cpaulgilman opened the pull request:
## Pull Request Template

### Description

Fix UI checks for critical load length to work with Custom Generation Battery configurations.

See #2048 for test procedure and sample file.

Fixes #2048 

### Checklist
- [ ] requires help revision and I added that label
- [ ] adds, removes, modifies, or deletes variables in existing compute modules
- [ ] adds a new compute module
- [ ] changes defaults
- [X] I've tagged this PR to a milestone

### _Reminders- this section can be deleted_
[Checking for PySAM Incompatible API Changes](https://github.com/NREL/SAM/wiki/PySAM-Incompatible-API-Changes-&-Regenerating-PySAM-Files).

[When do the PySAM files need to be regenerated?](https://github.com/NREL/SAM/wiki/PySAM-Incompatible-API-Changes-&-Regenerating-PySAM-Files#when-do-the-pysam-files-need-to-be-regenerated-via-export_config)

cpaulgilman committed:
Fix UI callback `crit_load` check for Custom Generation Battery configs

cpaulgilman committed:
Merge branch 'develop' into sam-2048-crit-load-error

cpaulgilman committed:
Remove debug message box.

brtietz commented on deploy/runtime/ui/Electric Load Data.json:
```diff
@@ -5824,11 +5824,15 @@
+        "\tis_critical_load_tech = ( tech == 'PV Battery' \r",
+        "\t  || tech == 'Custom Generation Battery'\r",
+        "\t  || tech == 'Standalone Battery' \r",
+        "\t  || tech == 'Generic Battery' \r",
```
Is there a reason Generic Battery is still in this list?

brtietz reviewed (commented):
The changes fix the bug, but I think there might be a little more work to do to clean up lingering references to generic battery. In addition to the one here, a 'find in files' search turned up suspicious references to Generic Battery in runtime/reports/pvbattery.samreport

cpaulgilman committed:
Remove "Generic Battery" references

cpaulgilman committed:
Change "Generic Battery" to "Custom Generation Battery" in report template

Note that pvbattery.samreport template does not support Custom Generation Battery, so this does not affect any functionality.

cpaulgilman commented:
> Is there a reason Generic Battery is still in this list?

No. That was an oversight. Thanks for pointing it out.

> ...suspicious references to Generic Battery in runtime/reports/pvbattery.samreport

I took care of that one -- it doesn't affect any functionality because the report template is not configured for the Generic Battery configurations. Additional work on the template is required to support PDF reports for Custom Generation Battery configurations, but I did change the references to "Generic Battery" to "Custom Generation Battery" in the scripts.

brtietz reviewed (approved):
Thanks for fixing!
